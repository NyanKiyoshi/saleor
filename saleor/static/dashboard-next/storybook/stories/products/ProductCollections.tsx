import { storiesOf } from "@storybook/react";
import * as React from "react";

import ProductCollections from "../../../product/components/ProductCollections";

const collections = [
  { id: "1", name: "Winter collection" },
  { id: "2", name: "Emperor's choice" }
];

storiesOf("Products / ProductCollections", module)
  .add("with no collections", () => (
    <ProductCollections collections={[]} onRowClick={() => {}} />
  ))
  .add("with collections", () => (
    <ProductCollections collections={collections} onRowClick={() => {}} />
  ));
