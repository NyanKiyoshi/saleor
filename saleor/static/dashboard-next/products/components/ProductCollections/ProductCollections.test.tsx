import * as React from "react";
import * as renderer from "react-test-renderer";

import ProductCollections from "./";

const collections = [
  { id: "1", name: "Winter collection" },
  { id: "2", name: "Emperor's choice" }
];

describe("<ProductCollections />", () => {
  it("renders when loading data", () => {
    const component = renderer.create(
      <ProductCollections onRowClick={jest.fn()} />
    );
    expect(component).toMatchSnapshot();
  });
  it("renders with data loaded", () => {
    const component = renderer.create(
      <ProductCollections onRowClick={jest.fn()} collections={collections} />
    );
    expect(component).toMatchSnapshot();
  });
});
