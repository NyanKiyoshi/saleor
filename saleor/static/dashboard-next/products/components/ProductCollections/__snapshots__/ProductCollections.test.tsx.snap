// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProductCollections /> renders when loading data 1`] = `
<div
  className="MuiPaper-root-3 MuiPaper-elevation2-7 MuiPaper-rounded-4 Component-card-1"
>
  <div
    className="Component-root-33"
  >
    <h2
      className="MuiTypography-root-36 MuiTypography-title-42 Component-title-35"
    >
      Collections
    </h2>
    <div
      className="Component-action-30"
    />
  </div>
  <table
    className="MuiTable-root-60"
  >
    <tbody
      className="MuiTableBody-root-61"
    >
      <tr
        className="MuiTableRow-root-62"
      >
        <td
          aria-sort={null}
          className="MuiTableCell-root-67 MuiTableCell-body-69"
          scope={undefined}
        >
          <span
            className="Component-skeleton-75"
            style={undefined}
          >
            â€Œ
          </span>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`<ProductCollections /> renders with data loaded 1`] = `
<div
  className="MuiPaper-root-3 MuiPaper-elevation2-7 MuiPaper-rounded-4 Component-card-1"
>
  <div
    className="Component-root-33"
  >
    <h2
      className="MuiTypography-root-36 MuiTypography-title-42 Component-title-35"
    >
      Collections
    </h2>
    <div
      className="Component-action-30"
    />
  </div>
  <table
    className="MuiTable-root-60"
  >
    <tbody
      className="MuiTableBody-root-61"
    >
      <tr
        className="MuiTableRow-root-62"
      >
        <td
          aria-sort={null}
          className="MuiTableCell-root-67 MuiTableCell-body-69 Component-link-2"
          onClick={undefined}
          scope={undefined}
        >
          Winter collection
        </td>
      </tr>
      <tr
        className="MuiTableRow-root-62"
      >
        <td
          aria-sort={null}
          className="MuiTableCell-root-67 MuiTableCell-body-69 Component-link-2"
          onClick={undefined}
          scope={undefined}
        >
          Emperor's choice
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
